lte/gateway/c/core/oai/test/s1ap_task/test_s1ap_mme_handlers_with_injected_state.cpp

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright 2021 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <gtest/gtest.h>
+#include <thread>
+
+#include "lte/gateway/c/core/oai/test/mock_tasks/mock_tasks.h"
+
+extern "C" {
+#include "lte/gateway/c/core/common/dynamic_memory_check.h"
+#include "lte/gateway/c/core/oai/common/log.h"
+#include "lte/gateway/c/core/oai/include/mme_config.h"
+#include "lte/gateway/c/core/oai/include/s1ap_state.h"
+#include "lte/gateway/c/core/oai/lib/bstr/bstrlib.h"
+#include "lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme.h"
+#include "lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_decoder.h"
+#include "lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.h"
+#include "lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_nas_procedures.h"
+}
+
+#include "lte/gateway/c/core/oai/test/s1ap_task/s1ap_mme_test_utils.h"
+#include "lte/gateway/c/core/oai/test/s1ap_task/mock_s1ap_op.h"
+#include "lte/gateway/c/core/oai/tasks/s1ap/s1ap_state_manager.h"
+
+extern bool hss_associated;
+
+namespace magma {
+namespace lte {
+
+task_zmq_ctx_t task_zmq_ctx_main_s1ap_with_injected_states;
+
+static int handle_message(zloop_t* loop, zsock_t* reader, void* arg) {
+  MessageDef* received_message_p = receive_msg(reader);
+
+  switch (ITTI_MSG_ID(received_message_p)) {
```

Comment:
Why is this empty? May be add a TODO if its for future use?

Reply:
Is this Comment addressed?

Reply:
Hi @pruthvihebbani. From my understanding, the role of this function is to mock the message handler function. Currently, in our test, we don't use this handler. Therefore, you see it empty. Indeed, in the future, if we need some functionalities related to the message handler, we should add them here. I add a TODO to remind us in the future, as your suggestion.